### PR TITLE
[Test-Proxy] Update Exception Handling For Error Propogation, Allow No-Op BodyKeySanitizer

### PR DESCRIPTION
**Pull request by scbedd**
@JoshLove-msft I will use this basis to propogate the TestMismatchExceptions back.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/Test.RecordEntries/post_delete_get_content.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         "x-ms-date": "Tue, 18 May 2021 23:27:42 GMT",
         "x-ms-version": "2019-02-02"
       },
-      "RequestBody": "{\u0022TableName\u0022: \u0022listtable09bf2a3d\u0022}",
+      "RequestBody": "{\u0022TableName\u0022:    \u0022listtable09bf2a3d\u0022}",
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/Test.RecordEntries/post_delete_get_content.json:20]: This is deliberate. I do this so I can easily check for a re-serialized body.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/HttpException.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        public HttpException(HttpStatusCode statusCode)
+        {
+            this.StatusCode = statusCode;
+        }
+
+        public HttpException(int httpStatusCode)
+            : this((HttpStatusCode)httpStatusCode)
+        {
+        }
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/HttpException.cs:27]:
```suggestion
                this.StatusCode = statusCode;        
        }
```

**Review by mikeharder — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Sanitizers/BodyKeySanitizer.cs [hunk 3]
```diff
@@ -53,15 +54,22 @@ public override string SanitizeTextBody(string contentType, string body)
...
+                    if(originalValue != replacement)
+                    {
+                        sanitized = true;
+                    }
                 }
             }
-            
-            return JsonConvert.SerializeObject(jsonO, SerializerSettings);
+
+            return sanitized ? JsonConvert.SerializeObject(jsonO, SerializerSettings) : body;
```
Comment (JoshLove-msft) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Sanitizers/BodyKeySanitizer.cs:72]: For some reason I didn't get a notification about this PR. It looks great! The only thing missing is the try/catch in this method.

**Pull request merged**